tests/python/relay/test_tidl.py

```diff
@@ -0,0 +1,210 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Unit tests for TIDL compilation."""
+import os
+import sys
+
+import argparse
+parser = argparse.ArgumentParser(description="Testing TIDL code generation")
+parser.add_argument("--batch_size", type=int, default=1, help="batch size")
+parser.add_argument("--num_tidl_subgraphs", type=int, default=1, help="number of TIDL subgraphs")
+parser.add_argument("--force_arm_only", help="Force ARM-only execution", action='store_true', default=False)
+
+try:
+    args = parser.parse_args()
+except SystemExit:
+    quit()
+
+batch_size = args.batch_size
+num_tidl_subgraphs = args.num_tidl_subgraphs
+force_arm_only = args.force_arm_only
+
+if os.getenv("ARM_GCC_PATH") is None:
+  sys.exit("Environment variable ARM_GCC_PATH is not set!")
+else: 
+  arm_gcc_path = os.getenv("ARM_GCC_PATH")
+arm_gcc = os.path.join(arm_gcc_path, "arm-linux-gnueabihf-g++")
+target  = "llvm -target=armv7l-linux-gnueabihf"
+
+import numpy as np
+import onnx
+import tvm
+import tvm.relay.testing
+from tvm import relay
+from tvm.contrib import cc
+from tvm.contrib import graph_runtime
+from tvm.contrib.download import download_testdata
+import mxnet as mx
+from mxnet import image
+from matplotlib import pyplot as plt
+import tensorflow as tf
+from tvm.relay.testing import tf as tf_testing
+from gluoncv import model_zoo, data, utils
+from gluoncv.data.transforms.presets.segmentation import test_transform
+from tvm.relay.backend.contrib import tidl
+
+def model_compile(model_name, mod_orig, params, 
+                  data_layout, input_node, input_data):
+    artifacts_folder = "./artifacts_" + model_name
+    if os.path.isdir(artifacts_folder):
+        filelist = [ f for f in os.listdir(artifacts_folder)]
+        for file in filelist:
+            os.remove(os.path.join(artifacts_folder, file))
+    else:
+        os.mkdir(artifacts_folder)
+
+    if force_arm_only is True:
+        mod = mod_orig
+        tidlEnabled = False
+    else:
+        tidl_compiler = tidl.TIDLCompiler("AM57", (6,3), 
+                                          num_tidl_subgraphs = num_tidl_subgraphs, 
+                                          data_layout = data_layout, 
+                                          artifacts_folder = artifacts_folder) 
+        input = {input_node: input_data}
+        mod = tidl_compiler.enable(mod_orig, params, input)

```

Comment:
We will need a way for this to return early before calibration since CI doesn't have the calibration tools. Should we add a 'test_mode' argument to TIDLCompiler?

Reply:
A second return variable was added to indicate that import and calibration are skipped because tools do not exist.